src/rgw/rgw_sal_daos.cc

```diff
@@ -374,7 +374,99 @@ int DaosUser::store_user(const DoutPrefixProvider* dpp, optional_yield y,
   return 0;
 }
 
+/*
+ * Tested DaosUser::remove_user with the following command lines:
+ *
+ * bin/radosgw-admin user create --uid johndoe --display-name "John Doe" --email johndoe@mail.com --no-mon-config
+ * bin/radosgw-admin user create --uid janedoe --display-name "Jane Doe" --email janedoe@mail.com --no-mon-config
+ * bin/radosgw-admin user create --uid joeuser --display-name "Joe User" --email joeuser@mail.com --no-mon-config
+ * bin/radosgw-admin user create --uid dgwuser --display-name "Dgw User" --email dgwuser@mail.com --no-mon-config
+ * bin/radosgw-admin user rm --uid johndoe
+ * bin/radosgw-admin user rm --uid janedoe
+ * bin/radosgw-admin user rm --uid joeuser
+ * bin/radosgw-admin user rm --uid dgwuser
+ */
 int DaosUser::remove_user(const DoutPrefixProvider* dpp, optional_yield y) {
+  const string name = info.user_id.to_str();
+
+  // the expectation is that the object version needs to be passed in as a method arg
+  //
+  // struct DaosUserInfo duinfo;
+  // int ret = store->read_user(dpp, USERS_DIR, name, &duinfo);
+  // obj_version obj_ver = duinfo.user_version;
+
+  // // Check if the user already exists
+  // if (ret == 0 && obj_ver.ver) {
+  //   // already exists.
+
+  //   if (objv_tracker.read_version.ver != obj_ver.ver) {
+  //     // Object version mismatch.. return ECANCELED
+  //     ret = -ECANCELED;
+  //     ldpp_dout(dpp, 0) << "User Read version mismatch read_version=" << objv_tracker.read_version.ver
+  //                       << " obj_ver=" << obj_ver.ver
+  //                       << dendl;
+  //     return ret;
+  //   }
+  //   obj_ver.ver++;
+  // } else {
+  //   obj_ver.ver = 1;
+  //   obj_ver.tag = "UserTAG";
+  // }
+
```

Comment:
Could you clean up this part? We could replicate [what DBStore does](https://github.com/zalsader/ceph/blob/82db5e0bdb4fc9ad742d3b11d76d81d2f332c272/src/rgw/store/dbstore/common/dbstore.cc#L378-L387).